fiat-roles/src/main/groovy/com/netflix/spinnaker/fiat/permissions/RedisPermissionsRepository.java

```diff
@@ -16,7 +16,228 @@
+            permission.getAccounts().stream().forEach(putInValuesMap);
+            break;
+          case APPLICATION:
+            permission.getApplications().stream().forEach(putInValuesMap);
+            break;
+        }
+
+        if (!resourceValues.isEmpty()) {
+          jedis.hmset(userKey(permission.getId(), r), resourceValues);
+          jedis.sadd(allUsersKey(), permission.getId());
+        }
+      }
+    } catch (Exception e) {
+      log.error("Storage exception writing " + permission.getId() + " entry.", e);
+    }
+    return this;
+  }
+
+  @Override
+  public UserPermission get(@NonNull String id) {
+    UserPermission userPermission = new UserPermission().setId(id);
+    try (Jedis jedis = jedisSource.getJedis()) {
+      for (Resource r : Resource.values()) {
+        Map<String, String> resourceMap = jedis.hgetAll(userKey(id, r));
+
+        switch (r) {
+          case ACCOUNT:
+            userPermission.getAccounts().addAll(extractAccounts(resourceMap));
+            break;
+          case APPLICATION:
+            userPermission.getApplications().addAll(extractApplications(resourceMap));
+            break;
+        }
+      }
+    } catch (Exception e) {
+      log.error("Storage exception reading " + id + " entry.", e);
+    }
+
+    return userPermission.isEmpty() ? null : userPermission;
+  }
+
+  @Override
+  public Map<String, UserPermission> getAllById() {
+    Table<String, Resource, Response<Map<String, String>>> responseTable = getAllFromRedis();
+    if (responseTable == null) {
+      return new HashMap<>(0);
+    }
+    Map<String, UserPermission> allById = new HashMap<>(responseTable.rowKeySet().size());
+
+    for (Map.Entry<String, Map<Resource, Response<Map<String, String>>>> entry1 : responseTable.rowMap().entrySet()) {
+      String userId = entry1.getKey();
+      for (Map.Entry<Resource, Response<Map<String, String>>> entry2 : entry1.getValue().entrySet()) {
+        Resource r = entry2.getKey();
+        Map<String /*resourceName*/, String /*resource json*/> resourceMap = entry2.getValue().get();
+
+        Function<String, UserPermission> newUser = id -> new UserPermission().setId(id);
+
+        switch (r) {
+          case ACCOUNT:
+            allById.computeIfAbsent(userId, newUser)
+                   .getAccounts()
+                   .addAll(extractAccounts(resourceMap));
+            break;
+          case APPLICATION:
+            allById.computeIfAbsent(userId, newUser)
+                   .getApplications()
+                   .addAll(extractApplications(resourceMap));
+        }
+      }
+    }
+    return allById;
+  }
+
+  private Table<String, Resource, Response<Map<String, String>>> getAllFromRedis() {
+    try (Jedis jedis = jedisSource.getJedis()) {
+      Set<String> allUserIds = jedis.smembers(allUsersKey());
+
+      Table<String, Resource, Response<Map<String, String>>> responseTable =
+          HashBasedTable.create(allUserIds.size(), Resource.values().length);
```

Comment:
If we find the binary is too large we can work to slim it down, but lets not prematurely optimize.